### ISSUE TITLE
False Positive for SSD_DO_NOT_USE_INSTANCE_LOCK_ON_SHARED_STATIC_DATA

### ISSUE THREAD

**Issue by jpschewe** [spotbugs/spotbugs#1978]
When locking on the class object the bug SSD_DO_NOT_USE_INSTANCE_LOCK_ON_SHARED_STATIC_DATA is reported. Locking on the class object should be the same as creating a static object to lock on as there is only a single instance of the class object.

```java
public class StaticMethodLock {
  
  private static int importdbCount = 0;

  public static int getNextDBCount() {
    synchronized (StaticMethodLock.class) {
      return importdbCount++;
    }
  }  

}
```

Reports
```
M C SSD: Static field "StaticMethodLock.getNextDBCount()" is modified by an instance level StaticMethodLock.getNextDBCount().  At StaticMethodLock.java:[line 7]
```

Using SpogBugs 4.6.0 with gradle plugin 5.0.5.
```
java -version
openjdk version "17.0.2" 2022-01-18
OpenJDK Runtime Environment (build 17.0.2+8-Ubuntu-120.04)
OpenJDK 64-Bit Server VM (build 17.0.2+8-Ubuntu-120.04, mixed mode, sharing)
```

**Comment by KengoTODA** [comment]
refs #1926

@gonczmisi could you check this?

**Comment by gonczmisi** [comment]
Sure, will take a look into it.

**Comment by KengoTODA** [comment]
closed by #1984
